junit/core/src/test/java/org/jboss/arquillian/junit/rules/RulesEnrichmentTestCase.java

```diff
@@ -61,7 +63,7 @@ protected void addExtensions(List<Class<?>> extensions)
     public void prepare()
     {
         Injector injector = InjectorImpl.of(getManager());
-        ServiceRegistry registry = new ServiceRegistry(injector);
+        ServiceRegistry registry = new ServiceRegistry(injector, new LinkedHashMap<Class<?>, Set<Class<?>>>());
```

Comment:
What you mention is done at `JavaSPIExtensionLoaderTestCase`.  This tests already existed before this change, it is just adapting to new contact. The idea is to not modify their purpose.